### PR TITLE
thunderx: simplified code of 052591 PMC emulation for clarity

### PR DESCRIPTION
**Pull request by jotego**
- removed unneeded loop counters
- removed unneeded multiplication/division operations
- matched variable names in both functions

Played the game for 20 minutes with no issues. Collision detection works the same.

### CODE REVIEW

File: src/mame/konami/thunderx.cpp [hunk 1]
```diff
@@ -367,10 +367,8 @@ void thunderx_state::run_collisions( int s0, int e0, int s1, int e1, int cm, int
 {
 	uint8_t* p0;
 	uint8_t* p1;
-	int ii, jj;
 
-	p0 = &m_pmcram[16 + 5 * s0];
-	for (ii = s0; ii < e0; ii++, p0 += 5)
+	for ( p0 = &m_pmcram[s0]; p0 < &m_pmcram[e0]; p0 += 5)
```
Comment (cuavas) [inline src/mame/konami/thunderx.cpp:371]: Please use “English” style spacing around parentheses (no space inside the parentheses) as the code was using previously.  It isn’t even consistent between the opening and closing parentheses now.

Also, you can reduce the scope of `p0` by declaring it in the `for` loop’s control expression.

File: src/mame/konami/thunderx.cpp [hunk 2]
```diff
@@ -384,8 +382,7 @@ void thunderx_state::run_collisions( int s0, int e0, int s1, int e1, int cm, int
 		t0 = p0[4] - p0[2];
 		b0 = p0[4] + p0[2];
 
-		p1 = &m_pmcram[16 + 5 * s1];
-		for (jj = s1; jj < e1; jj++,p1 += 5)
+		for ( p1 = &m_pmcram[s1]; p1 < &m_pmcram[e1]; p1 += 5)
```
Comment (cuavas) [inline src/mame/konami/thunderx.cpp:385]: Also, as above, use “English” spacing around parentheses, and you may as well reduce the scope of `p1` while you’re changing the code anyway.

File: src/mame/konami/thunderx.cpp [hunk 3]
```diff
@@ -441,30 +438,28 @@ void thunderx_state::calculate_collisions()
 	// hit mask is 40 to set bit on object 0 and object 1
 	// hit mask is 20 to set bit on object 1 only
 
-	Y0 = m_pmcram[0];
-	Y0 = (Y0 << 8) + m_pmcram[1];
-	Y0 = (Y0 - 15) / 5;
-	Y1 = (m_pmcram[2] - 15) / 5;
+	e0 = m_pmcram[0];
+	e0 = (e0 << 8) + m_pmcram[1];
+	e1 = m_pmcram[2];
```
Comment (cuavas) [inline src/mame/konami/thunderx.cpp:443]: If you’re changing this anyway, you may as well declare `e0` and `e1` at initialisation, initialise `e0` in a single statement, and make them both `const`.

File: src/mame/konami/thunderx.cpp [hunk 4]
```diff
@@ -441,30 +438,28 @@ void thunderx_state::calculate_collisions()
...
-		X0 = (m_pmcram[5] - 16) / 5;
-		X1 = (m_pmcram[6] - 16) / 5;
+		s0 = m_pmcram[5];
+		s1 = m_pmcram[6];
 	}
 
-	CM = m_pmcram[3];
-	HM = m_pmcram[4];
+	cm = m_pmcram[3];
+	hm = m_pmcram[4];
```
Comment (cuavas) [inline src/mame/konami/thunderx.cpp:460]: You may as well declare these at initialisation and make them `const` as well.

**Review by cuavas — CHANGES_REQUESTED** [review]

**Comment by jotego** [comment]
I have applied the changes requested

File: src/mame/konami/thunderx.cpp [hunk 5]
```diff
@@ -441,30 +433,27 @@ void thunderx_state::calculate_collisions()
 	// hit mask is 40 to set bit on object 0 and object 1
 	// hit mask is 20 to set bit on object 1 only
 
-	Y0 = m_pmcram[0];
-	Y0 = (Y0 << 8) + m_pmcram[1];
-	Y0 = (Y0 - 15) / 5;
-	Y1 = (m_pmcram[2] - 15) / 5;
+	const int e0 = (m_pmcram[0]<<8) | m_pmcram[1];
+	const int e1 = m_pmcram[2];
```
Comment (cuavas) [inline src/mame/konami/thunderx.cpp:437]: Note that this actually does change behaviour – previously it divided the Y1 value by 5 and then multiplied by 5 again later.  If the actual value was not a multiple of 5, the remainder would be truncated.  That’s no longer the case.  It now uses the actual value from RAM directly.

Comment (jotego) [reply src/mame/konami/thunderx.cpp:437]: Yes, but data is accessed in groups of 5 later on. The new code makes more sense and is more parsimonius. I have not been able to find any problem in the gameplay after the change.

I do not know why the original author ended up writting those divisions and multiplications. They might be some leftover from all the guess work needed to emulate the chip or maybe he was trying to make sense of the _program_ bytes written by the CPU.

**Pull request merged**